parser/Parser.cc

```diff
@@ -149,7 +149,6 @@ unique_ptr<Node> Parser::run(core::GlobalState &gs, core::FileRef file, Parser::
     // report the hints. Always reporting the original parse errors ensures that the user can always
     // see the real cause in case the hints are misleading.
     if (!driver->diagnostics.empty()) {
-        file.data(gs).setHasParseErrors(true);
```

Comment:
It's not necessary to remove this, but it did seem redundant with the processing in pipeline.cc at this point.

Reply:
I believe that this *does* become redundant with the other changes in this PR if you use the `GlobalState::_error` approach described in another comment.

If that's not the case, I'd love to dig into why it's not the case.